python/arcticdb/adapters/mongo_library_adapter.py

```diff
@@ -0,0 +1,102 @@
+"""
+Copyright 2023 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+from pickletools import int4
+import re
+import time
+from typing import Optional
+
+from arcticdb.options import LibraryOptions
+from arcticc.pb2.storage_pb2 import EnvironmentConfigsMap, LibraryConfig
+from arcticdb.version_store.helper import add_mongo_library_to_env
+from arcticdb.config import _DEFAULT_ENV
+from arcticdb.version_store._store import NativeVersionStore
+from arcticdb.adapters.arctic_library_adapter import ArcticLibraryAdapter, set_library_options
+from arcticdb_ext.storage import Library
+from arcticdb.encoding_version import EncodingVersion
+from collections import namedtuple
+from dataclasses import dataclass, fields
+
+PARSED_QUERY = namedtuple("PARSED_QUERY", ["region"])
+
+
+@dataclass
+class ParsedQuery:
+    min_pool_size: Optional[int] = 0
+    max_pool_size: Optional[int] = 0
+    selection_timeout_ms: Optional[int] = 0
+
+
+class MongoLibraryAdapter(ArcticLibraryAdapter):
+    REGEX = r"(?P<endpoint>mongodb://[^?]+)(?P<options>\?.*)?"
+
+    @staticmethod
+    def supports_uri(uri: str) -> bool:
+        return uri.startswith("mongodb://")
+
+    def __init__(self, uri: str, *args, **kwargs):
```

Comment:
Any way to use `pymongo` for this? Something like https://pymongo.readthedocs.io/en/stable/api/pymongo/uri_parser.html

Doing the query parsing well is harder than it looks, especially to give good validation and user facing errors...

Reply:
All that is being parsed are run-time config, so there's no need to parse the URI at the moment.

Reply:
It's still good to validate it early. And I think there will be a need to parse it actually, to separate out the connection details from the options.

Reply:
Not if we have to import an entire Python dependency to do it.

If we have to have this early check, it should call into C++ and use the C++ SDK we've already compiled in.